{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "rkFvOSrDlaSq",
        "outputId": "17ca29d2-3142-48e9-e799-a414669d2673"
      },
      "outputs": [
        {
          "ename": "",
          "evalue": "",
          "output_type": "error",
          "traceback": [
            "\u001b[1;31mFailed to start the Kernel. \n",
            "\u001b[1;31mTraceback (most recent call last):\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/pythonFiles/vscode_datascience_helpers/jupyter_daemon.py\", line 158, in _start_notebook\n",
            "\u001b[1;31m    from notebook import notebookapp as app\n",
            "\u001b[1;31mImportError: cannot import name 'notebookapp' from 'notebook' (/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/notebook/__init__.py)\n",
            "\u001b[1;31m\n",
            "\u001b[1;31mDuring handling of the above exception, another exception occurred:\n",
            "\u001b[1;31m\n",
            "\u001b[1;31mTraceback (most recent call last):\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/pythonFiles/vscode_datascience_helpers/daemon/daemon_python.py\", line 54, in _decorator\n",
            "\u001b[1;31m    return func(self, *args, **kwargs)\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/pythonFiles/vscode_datascience_helpers/jupyter_daemon.py\", line 111, in m_exec_module_observable\n",
            "\u001b[1;31m    self._start_notebook(args, cwd, env)\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/pythonFiles/vscode_datascience_helpers/jupyter_daemon.py\", line 160, in _start_notebook\n",
            "\u001b[1;31m    from notebook import app as app\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/notebook/app.py\", line 17, in <module>\n",
            "\u001b[1;31m    from jupyter_server.serverapp import flags\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/jupyter_server/serverapp.py\", line 39, in <module>\n",
            "\u001b[1;31m    from jupyter_events.logger import EventLogger\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/jupyter_events/__init__.py\", line 3, in <module>\n",
            "\u001b[1;31m    from .logger import EVENTS_METADATA_VERSION, EventLogger\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/jupyter_events/logger.py\", line 14, in <module>\n",
            "\u001b[1;31m    from jsonschema import ValidationError\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/jsonschema/__init__.py\", line 13, in <module>\n",
            "\u001b[1;31m    from jsonschema._format import FormatChecker\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/jsonschema/_format.py\", line 11, in <module>\n",
            "\u001b[1;31m    from jsonschema.exceptions import FormatError\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/jsonschema/exceptions.py\", line 14, in <module>\n",
            "\u001b[1;31m    from attrs import define\n",
            "\u001b[1;31mModuleNotFoundError: No module named 'attrs'\n",
            "\u001b[1;31m\n",
            "\u001b[1;31mFailed to run jupyter as observable with args notebook --no-browser --notebook-dir=\"/Users/matthewrobinson/Desktop/GitHub/dao2702x\" --config=/var/folders/5d/pf9tt81n14zc16c3ky9s8kj40000gn/T/a448a464-cdd1-4679-9ab7-6725e562f88d/jupyter_notebook_config.py --NotebookApp.iopub_data_rate_limit=10000000000.0. \n",
            "\u001b[1;31mView Jupyter <a href='command:jupyter.viewOutput'>log</a> for further details."
          ]
        }
      ],
      "source": [
        "# from google.colab import drive\n",
        "# import warnings\n",
        "# warnings.filterwarnings(\"ignore\", category=FutureWarning)\n",
        "\n",
        "# drive.mount('/content/drive')\n",
        "\n",
        "# Load the data and check the first few lines\n",
        "file_path = \"TMDB_movie_dataset_cleaned.csv\" # use your own path"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "YQBQRlxTlU_u",
        "outputId": "6b689192-ab1e-4ad7-f5ad-a9f29fa6014b"
      },
      "outputs": [
        {
          "ename": "",
          "evalue": "",
          "output_type": "error",
          "traceback": [
            "\u001b[1;31mFailed to start the Kernel. \n",
            "\u001b[1;31mTraceback (most recent call last):\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/pythonFiles/vscode_datascience_helpers/jupyter_daemon.py\", line 158, in _start_notebook\n",
            "\u001b[1;31m    from notebook import notebookapp as app\n",
            "\u001b[1;31mImportError: cannot import name 'notebookapp' from 'notebook' (/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/notebook/__init__.py)\n",
            "\u001b[1;31m\n",
            "\u001b[1;31mDuring handling of the above exception, another exception occurred:\n",
            "\u001b[1;31m\n",
            "\u001b[1;31mTraceback (most recent call last):\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/pythonFiles/vscode_datascience_helpers/daemon/daemon_python.py\", line 54, in _decorator\n",
            "\u001b[1;31m    return func(self, *args, **kwargs)\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/pythonFiles/vscode_datascience_helpers/jupyter_daemon.py\", line 111, in m_exec_module_observable\n",
            "\u001b[1;31m    self._start_notebook(args, cwd, env)\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/pythonFiles/vscode_datascience_helpers/jupyter_daemon.py\", line 160, in _start_notebook\n",
            "\u001b[1;31m    from notebook import app as app\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/notebook/app.py\", line 17, in <module>\n",
            "\u001b[1;31m    from jupyter_server.serverapp import flags\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/jupyter_server/serverapp.py\", line 39, in <module>\n",
            "\u001b[1;31m    from jupyter_events.logger import EventLogger\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/jupyter_events/__init__.py\", line 3, in <module>\n",
            "\u001b[1;31m    from .logger import EVENTS_METADATA_VERSION, EventLogger\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/jupyter_events/logger.py\", line 14, in <module>\n",
            "\u001b[1;31m    from jsonschema import ValidationError\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/jsonschema/__init__.py\", line 13, in <module>\n",
            "\u001b[1;31m    from jsonschema._format import FormatChecker\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/jsonschema/_format.py\", line 11, in <module>\n",
            "\u001b[1;31m    from jsonschema.exceptions import FormatError\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/jsonschema/exceptions.py\", line 14, in <module>\n",
            "\u001b[1;31m    from attrs import define\n",
            "\u001b[1;31mModuleNotFoundError: No module named 'attrs'\n",
            "\u001b[1;31m\n",
            "\u001b[1;31mFailed to run jupyter as observable with args notebook --no-browser --notebook-dir=\"/Users/matthewrobinson/Desktop/GitHub/dao2702x\" --config=/var/folders/5d/pf9tt81n14zc16c3ky9s8kj40000gn/T/a448a464-cdd1-4679-9ab7-6725e562f88d/jupyter_notebook_config.py --NotebookApp.iopub_data_rate_limit=10000000000.0. \n",
            "\u001b[1;31mView Jupyter <a href='command:jupyter.viewOutput'>log</a> for further details."
          ]
        }
      ],
      "source": [
        "# Importing necessary libraries\n",
        "import pandas as pd\n",
        "import numpy as np\n",
        "import matplotlib.pyplot as plt\n",
        "import seaborn as sns\n",
        "\n",
        "# Load the dataset\n",
        "df = pd.read_csv(file_path)\n",
        "\n",
        "# Setting general visualization aesthetics\n",
        "sns.set_style('whitegrid')\n",
        "\n",
        "# Filter the DataFrame to include only released movies with revenue > 0\n",
        "df = df[(df['status'] == 'Released') & (df['revenue'] > 0)]\n",
        "print(len(df))\n",
        "\n",
        "# Calculate profit\n",
        "df['profit'] = df['revenue'] - df['budget']\n",
        "\n",
        "# Process genres\n",
        "df_genres_expanded = df['genres'].str.get_dummies(sep=', ')\n",
        "df_combined = pd.concat([df, df_genres_expanded], axis=1)\n",
        "\n",
        "# Calculate average profit for each genre\n",
        "genre_average_profit = df_genres_expanded.mul(df_combined['profit'], axis=0).mean().sort_values(ascending=False)\n",
        "\n",
        "# Top 10 Genres by Average Profit\n",
        "plt.figure(figsize=(12, 8))\n",
        "top_10_genres = genre_average_profit.head(10)\n",
        "top_10_genres.plot(kind='bar', color='green')\n",
        "plt.title('Top 10 Genres by Average Profit')\n",
        "plt.xlabel('Genre')\n",
        "plt.ylabel('Average Profit ($)')\n",
        "plt.xticks(rotation=45)\n",
        "plt.show()\n",
        "\n",
        "# Displaying the top genres by average profit in the console\n",
        "print(\"Top 10 Genres by Average Profit:\")\n",
        "print(top_10_genres)\n",
        "\n",
        "# Average revenue by language\n",
        "language_revenue = df.groupby('original_language')['revenue'].mean().sort_values(ascending=False)\n",
        "\n",
        "# Top 10 Languages by Average Revenue\n",
        "plt.figure(figsize=(12, 8))\n",
        "top_10_languages = language_revenue.head(10)\n",
        "top_10_languages.plot(kind='bar', color='skyblue')\n",
        "plt.title('Top 10 Languages by Average Revenue')\n",
        "plt.xlabel('Original Language')\n",
        "plt.ylabel('Average Revenue ($)')\n",
        "plt.xticks(rotation=45)\n",
        "plt.show()\n",
        "\n",
        "# Displaying the top languages by average revenue in the console\n",
        "print(\"Top 10 Languages by Average Revenue:\")\n",
        "print(top_10_languages)\n",
        "\n",
        "\n",
        "# Count of movies created by language\n",
        "language_counts = df['original_language'].value_counts()\n",
        "\n",
        "# Top 10 Languages by Count of Movies\n",
        "plt.figure(figsize=(12, 8))\n",
        "top_10_languages_count = language_counts.head(10)\n",
        "top_10_languages_count.plot(kind='bar', color='orange')\n",
        "plt.title('Top 10 Languages by Count of Movies')\n",
        "plt.xlabel('Original Language')\n",
        "plt.ylabel('Count of Movies')\n",
        "plt.xticks(rotation=45)\n",
        "plt.show()\n",
        "\n",
        "# Displaying the count of movies by language in the console\n",
        "print(\"Top 10 Languages by Count of Movies:\")\n",
        "print(top_10_languages_count)\n",
        "\n",
        "# Identify top 10 languages by average revenue for box plot\n",
        "top_10_language_names = top_10_languages.index.tolist()\n",
        "\n",
        "# Filter DataFrame for the top 10 languages\n",
        "df_top_languages = df[df['original_language'].isin(top_10_language_names)]\n",
        "\n",
        "# Calculate average revenue for sorting purposes\n",
        "average_revenue_sorted = df_top_languages.groupby('original_language')['revenue'].mean().sort_values(ascending=False)\n",
        "\n",
        "# Box plot of revenue by top 10 languages, sorted by average revenue\n",
        "plt.figure(figsize=(14, 8))\n",
        "sns.boxplot(x='original_language', y='revenue', data=df_top_languages, order=average_revenue_sorted.index)\n",
        "plt.title('Box Plot of Revenue by Top 10 Languages (Sorted by Average Revenue)')\n",
        "plt.xlabel('Original Language')\n",
        "plt.ylabel('Revenue ($)')\n",
        "plt.xticks(rotation=45)\n",
        "plt.yscale('log')  # Optional: use a logarithmic scale for better visibility of revenue distribution\n",
        "plt.show()\n",
        "\n",
        "# Filter for English movies\n",
        "english_movies = df[df['original_language'] == 'en']\n",
        "korean_movies = df[df['original_language'] == 'ko']\n",
        "\n",
        "# Calculate the threshold for low revenue\n",
        "# For this example, we will use the median revenue as the threshold\n",
        "revenue_threshold = english_movies['revenue'].median()\n",
        "revenue_threshold_ko = korean_movies['revenue'].median()\n",
        "\n",
        "# Count the number of English movies below the revenue threshold\n",
        "low_revenue_movies = english_movies[english_movies['revenue'] < revenue_threshold]\n",
        "low_revenue_movies_ko = korean_movies[korean_movies['revenue'] < revenue_threshold]\n",
        "number_of_low_revenue_movies = len(low_revenue_movies)\n",
        "number_of_low_revenue_movies_ko = len(low_revenue_movies_ko)\n",
        "\n",
        "# Display the results\n",
        "print(f\"Number of English movies with low revenue (below median of ${revenue_threshold:.2f}): {number_of_low_revenue_movies}\")\n",
        "print(low_revenue_movies[['title', 'revenue']])\n",
        "print(f\"Number of Korean movies with low revenue (below median of ${revenue_threshold:.2f}): {number_of_low_revenue_movies_ko}\")\n",
        "print(low_revenue_movies_ko[['title', 'revenue']])"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "309yL8NuvNJH"
      },
      "outputs": [
        {
          "ename": "",
          "evalue": "",
          "output_type": "error",
          "traceback": [
            "\u001b[1;31mFailed to start the Kernel. \n",
            "\u001b[1;31mTraceback (most recent call last):\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/pythonFiles/vscode_datascience_helpers/jupyter_daemon.py\", line 158, in _start_notebook\n",
            "\u001b[1;31m    from notebook import notebookapp as app\n",
            "\u001b[1;31mImportError: cannot import name 'notebookapp' from 'notebook' (/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/notebook/__init__.py)\n",
            "\u001b[1;31m\n",
            "\u001b[1;31mDuring handling of the above exception, another exception occurred:\n",
            "\u001b[1;31m\n",
            "\u001b[1;31mTraceback (most recent call last):\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/pythonFiles/vscode_datascience_helpers/daemon/daemon_python.py\", line 54, in _decorator\n",
            "\u001b[1;31m    return func(self, *args, **kwargs)\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/pythonFiles/vscode_datascience_helpers/jupyter_daemon.py\", line 111, in m_exec_module_observable\n",
            "\u001b[1;31m    self._start_notebook(args, cwd, env)\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/.vscode/extensions/ms-toolsai.jupyter-2022.11.1003412109/pythonFiles/vscode_datascience_helpers/jupyter_daemon.py\", line 160, in _start_notebook\n",
            "\u001b[1;31m    from notebook import app as app\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/notebook/app.py\", line 17, in <module>\n",
            "\u001b[1;31m    from jupyter_server.serverapp import flags\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/jupyter_server/serverapp.py\", line 39, in <module>\n",
            "\u001b[1;31m    from jupyter_events.logger import EventLogger\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/jupyter_events/__init__.py\", line 3, in <module>\n",
            "\u001b[1;31m    from .logger import EVENTS_METADATA_VERSION, EventLogger\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/jupyter_events/logger.py\", line 14, in <module>\n",
            "\u001b[1;31m    from jsonschema import ValidationError\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/jsonschema/__init__.py\", line 13, in <module>\n",
            "\u001b[1;31m    from jsonschema._format import FormatChecker\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/jsonschema/_format.py\", line 11, in <module>\n",
            "\u001b[1;31m    from jsonschema.exceptions import FormatError\n",
            "\u001b[1;31m  File \"/Users/matthewrobinson/Library/Python/3.10/lib/python/site-packages/jsonschema/exceptions.py\", line 14, in <module>\n",
            "\u001b[1;31m    from attrs import define\n",
            "\u001b[1;31mModuleNotFoundError: No module named 'attrs'\n",
            "\u001b[1;31m\n",
            "\u001b[1;31mFailed to run jupyter as observable with args notebook --no-browser --notebook-dir=\"/Users/matthewrobinson/Desktop/GitHub/dao2702x\" --config=/var/folders/5d/pf9tt81n14zc16c3ky9s8kj40000gn/T/a448a464-cdd1-4679-9ab7-6725e562f88d/jupyter_notebook_config.py --NotebookApp.iopub_data_rate_limit=10000000000.0. \n",
            "\u001b[1;31mView Jupyter <a href='command:jupyter.viewOutput'>log</a> for further details."
          ]
        }
      ],
      "source": [
        "# 1. Calculate average revenue per language and sort by language name\n",
        "language_revenue = df.groupby('original_language')['revenue'].mean().sort_index()\n",
        "\n",
        "# Filter to the top 10 languages by average revenue\n",
        "top_languages_revenue = language_revenue.nlargest(10)\n",
        "\n",
        "# Plotting the average revenue by language (horizontal bar plot)\n",
        "plt.figure(figsize=(12, 8))\n",
        "top_languages_revenue.plot(kind='barh', color='skyblue')\n",
        "plt.title('Top 10 Languages by Average Revenue')\n",
        "plt.xlabel('Average Revenue')\n",
        "plt.ylabel('Original Language')\n",
        "plt.show()\n",
        "\n",
        "# 2. Calculate average profit margin per language and sort by language name\n",
        "language_profit_margin = df.groupby('original_language')['profit_margin'].mean().sort_index()\n",
        "\n",
        "# Filter to the top 10 languages by average profit margin\n",
        "top_languages_profit_margin = language_profit_margin.nlargest(10)\n",
        "\n",
        "# Plotting the average profit margin by language (horizontal bar plot)\n",
        "plt.figure(figsize=(12, 8))\n",
        "top_languages_profit_margin.plot(kind='barh', color='lightgreen')\n",
        "plt.title('Top 10 Languages by Average Profit Margin')\n",
        "plt.xlabel('Average Profit Margin')\n",
        "plt.ylabel('Original Language')\n",
        "plt.show()\n"
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "name": "python",
      "version": "3.10.2 (v3.10.2:a58ebcc701, Jan 13 2022, 14:50:16) [Clang 13.0.0 (clang-1300.0.29.30)]"
    },
    "vscode": {
      "interpreter": {
        "hash": "949777d72b0d2535278d3dc13498b2535136f6dfe0678499012e853ee9abcab1"
      }
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}
